plugins/modules/dns_record.py

```diff
@@ -0,0 +1,605 @@
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = RecordApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            if self.params["zone"] is not None:
+                filter = f"zone=='{self.params['zone']}' and type=='{self.params['type']}'"
+                resp = RecordApi(self.client).list(filter=filter, inherit="full")
+
+                # self.fail_json(msg=f"Found multiple Record: {resp.results}")
+
+                if len(resp.results) == 1:
+                    return resp.results[0]
+                if len(resp.results) == 0:
+                    return None
+
+                for i in resp.results:
+                    if i["address"] != self.params["address"]:
+                        resp.results.pop(i)
+
+                if len(resp.results) == 1:
+                    return resp.results[0]
+                if len(resp.results) > 1:
+                    self.fail_json(msg=f"Found multiple Record: {resp.results}")
+                if len(resp.results) == 0:
+                    return None
+
+            elif self.params["absolute_name_spec"] is not None:
+
+                filter = f"absolute_name_spec=='{self.params['absolute_name_spec']}' and type=='{self.params['type']}'"
+                resp = RecordApi(self.client).list(filter=filter, inherit="full")
+
+                # self.fail_json(msg=f"Found multiple Record: {resp.results}")
+
+                if len(resp.results) == 1:
+                    return resp.results[0]
+                if len(resp.results) == 0:
+                    return None
+
+                for i in resp.results:
+                    if i["address"] != self.params["address"]:
+                        resp.results.pop(i)
+
+                if len(resp.results) == 1:
+                    return resp.results[0]
+
+                for i in resp.results:
+                    if i["view"] != self.params["view"]:
+                        resp.results.pop(i)
+
+                if len(resp.results) == 1:
+                    return resp.results[0]
+                if len(resp.results) > 1:
+                    self.fail_json(msg=f"Found multiple Record: {resp.results}")
+                if len(resp.results) == 0:
+                    return None
+
+            else:
+                return self.fail_json(msg="Either Zone or ANS or Name in Zone is required to create a record")
```

Comment:
```suggestion
                return self.fail_json(msg="Either Zone or Name in Zone and Zone or Absolute Name Spec and View is required to create a record")
```

Use full forms please. Refrain from using abbreviations like NIZ

Reply:
fixed